Assets/MRTK/Core/Utilities/BuildAndDeploy/UwpAppxBuildTools.cs

```diff
@@ -90,13 +90,27 @@ public static async Task<bool> BuildAppxAsync(UwpBuildInfo buildInfo, Cancellati
             string storagePath = Path.GetFullPath(Path.Combine(Path.Combine(Application.dataPath, ".."), buildInfo.OutputDirectory));
             string solutionProjectPath = Path.GetFullPath(Path.Combine(storagePath, $@"{PlayerSettings.productName}.sln"));
 
+            int exitCode;
+
             // Building the solution requires first restoring NuGet packages - when built through
             // Visual Studio, VS does this automatically - when building via msbuild like we're doing here,
             // we have to do that step manually.
-            int exitCode = await Run(msBuildPath,
+            // We use msbuild for nuget restore by default, but if a path to nuget.exe is supplied then we use that executable
+            if (string.IsNullOrEmpty(buildInfo.NugetExecutablePath))
+            {
+                exitCode = await Run(msBuildPath,
                 $"\"{solutionProjectPath}\" /t:restore {GetMSBuildLoggingCommand(buildInfo.LogDirectory, "nugetRestore.log")}",
                 !Application.isBatchMode,
                 cancellationToken);
+            }
+            else
+            {
+                exitCode = await Run(buildInfo.NugetExecutablePath,
```

Comment:
this works but didn't you want to generalize this a bit more and just invoke an arbitrary optional callback with the `solutionProjectPath` (which the provider of the callback could use to invoke nuget.exe if they desire)? Either way works though.